packages/client/src/ontology/OntologyProvider.ts

```diff
@@ -32,7 +32,7 @@ export interface OntologyProvider {
    */
   getObjectDefinition: (
     apiName: string,
-  ) => Promise<ObjectTypeDefinition<any>>;
+  ) => Promise<ObjectTypeDefinition<any> & { rid: string }>;
```

Comment:
Just double checking, we're only adding rid here for internal use when we want to get object defs for batch links right?

Reply:
Correct. Thats why I didn't add it to the actual `ObjectTypeDefinition`, especially as thats in the api package and shared with `legacy-client`. We should not need this in an ideal world.